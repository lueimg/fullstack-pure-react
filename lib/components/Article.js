import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ArticleWrapper = styled.div`
    padding-bottom: 10px;
    border-bottom: 1px solid #aaa;
    margin-bottom: 10px;
`;
const Title = styled.div`
    font-weight: bold
`;
const DateWrapper = styled.div`
    font-size: 0.85em;
    color: #888
`;
const Author = styled.div`
    padding-top: 10px;
    padding-bottom: 10px;
`;
const Body = styled.div`
    padding-left: 20px;
`;

const dateDisplay = (date) => {
    return new Date(date).toDateString();
};

class Article extends Component {
   
    render() {
        const {article} = this.props;
        const {store} = this.context;

        const author = store.lookupAuthor(article.authorId);

        return (
            <ArticleWrapper>
                <Title>{article.title}</Title>
                <DateWrapper>{dateDisplay(article.date)}</DateWrapper>
                <Author>
                    <a href={author.website}>
                        {author.firstName} - {author.lastName}
                    </a>
                </Author>
                <Body>{article.body}</Body>
            </ArticleWrapper>
        );
    }
}

Article.propTypes = {
    article: PropTypes.shape({
        title: PropTypes.string.isRequired,
        body: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired
    })
};

Article.contextTypes = {
    store: PropTypes.object
};

export default Article;