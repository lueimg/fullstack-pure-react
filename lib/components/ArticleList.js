import React, { Component } from 'react';
import Article from './Article';

export class ArticleList extends Component {
    render() {
        return (
            <div>
                {Object.values(this.props.articles).map((article) => 
                    <Article 
                        key={article.id}
                        article={article}
                        
                    />)}
            </div>
        );
    }
}

export default ArticleList; 