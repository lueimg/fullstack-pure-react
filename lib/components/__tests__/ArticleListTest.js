import React from 'react';
import ArticleList from '../ArticleList';
// import renderer from 'react-test-renderer';
import 'babel-polyfill';
import { shallow } from 'enzyme';
import Article from '../Article';
// remove warning props
Article.propTypes = {};

describe('ArticleList', () => {
    const testProps = {
        articles: {
            a: { id: 'a' },
            b: { id: 'b' },
        },
        store: {
            lookupAuthor: jest.fn(() => ({})),
        },
    };

    it('renders correctly', () => {
        const wrapper = shallow(
            <ArticleList 
                {...testProps}
            />
        );

        expect(wrapper.node.props.children.length).toBe(2);
        // console.log(tree);
        // expect(tree.children.length).toBe(2);
        expect(wrapper).toMatchSnapshot();

    });

});