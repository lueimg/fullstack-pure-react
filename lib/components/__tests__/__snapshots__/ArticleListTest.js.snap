// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    className="sc-bdVaJa iKbXFR"
  >
    <div
      className="sc-bwzfXH kINcIv"
    />
    <div
      className="sc-htpNat kclBkK"
    >
      Invalid Date
    </div>
    <div
      className="sc-bxivhb hbQRAT"
    >
      <a
        href={undefined}
      >
         - 
      </a>
    </div>
    <div
      className="sc-ifAKCX QLKmc"
    />
  </div>
  <div
    className="sc-bdVaJa iKbXFR"
  >
    <div
      className="sc-bwzfXH kINcIv"
    />
    <div
      className="sc-htpNat kclBkK"
    >
      Invalid Date
    </div>
    <div
      className="sc-bxivhb hbQRAT"
    >
      <a
        href={undefined}
      >
         - 
      </a>
    </div>
    <div
      className="sc-ifAKCX QLKmc"
    />
  </div>
</div>
`;
